From 52794d4e7f0ab58ad7f22657846ea2274279af28 Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Fri, 29 Jul 2016 07:51:23 -0500
Subject: [PATCH 4/4] KLUDGE: More workaround weak alias, this time tackling pthread stuff.

In theory we'll now pull in pthread stuff when it's not strictly
needed, but that seems okay for now.
---
 src/thread/pthread_create.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/thread/pthread_create.c b/src/thread/pthread_create.c
index 9f6b98e..4513f47 100644
--- a/src/thread/pthread_create.c
+++ b/src/thread/pthread_create.c
@@ -157,10 +157,12 @@ static int start_c11(void *p)
 #define ROUND(x) (((x)+PAGE_SIZE-1)&-PAGE_SIZE)
 
 /* pthread_key_create.c overrides this */
-static volatile size_t dummy = 0;
-weak_alias(dummy, __pthread_tsd_size);
-static void *dummy_tsd[1] = { 0 };
-weak_alias(dummy_tsd, __pthread_tsd_main);
+//static volatile size_t dummy = 0;
+//weak_alias(dummy, __pthread_tsd_size);
+extern volatile int  __pthread_tsd_size;
+//static void *dummy_tsd[1] = { 0 };
+//weak_alias(dummy_tsd, __pthread_tsd_main);
+extern void *__pthread_tsd_main;
 
 volatile int __block_new_threads = 0;
 
-- 
1.7.1

